certmgr/mgr/file.go

```diff
@@ -201,6 +215,20 @@ func ReadSpecFile(path string, defaults *ParsableSpecOptions) (*cert.Spec, error
 	// transfer the parsed durations into their final resting spot.
 	spec.FinalizeSpecOptionParsing()
 
+	if len(spec.ParsedKeyUsages) > 0 {
+		for _, KeyUsageRaw := range spec.ParsedKeyUsages {
+			keyUsage, ok := config.ExtKeyUsage[strings.ToLower(KeyUsageRaw)]
+			if !ok {
+				return nil, fmt.Errorf("spec %s specifies unknown key usage '%s'. Valid values are: [%v]", path, KeyUsageRaw, validExtUsage)
```

Comment:
Would it be better to use `%q` with `[]string`?

Reply:
Yes: https://goplay.space/#qzee2SZPNt7

Reply:
Pushed a new commit fixing this

Reply:
Sorry, I could have provided a better example. What we meant was you don't need to worry about any of the string formatting you're doing in `validExtUsage` if you return a `[]string` and use `%q`. Here's a full example:
```go
package main

import (
	"fmt"

	"github.com/cloudflare/cfssl/config"
)

func validExtUsages() []string {
	usages := make([]string, 0, len(config.ExtKeyUsage))
	for usage := range config.ExtKeyUsage {
		usages = append(usages, usage)
	}
	return usages
}

func main() {
	fmt.Printf("%q\n", validExtUsages())
}
```

Reply:
Oh I see what you mean, updated again to return a []string,  and also to just define validExtUsages as a function instead of a string that is built from a function at compile time -- the way I had done it is more efficient, but this code doesn't get called enough for that to matter.